libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClientFactory.kt

```diff
@@ -64,6 +64,7 @@ class RustMatrixClientFactory @Inject constructor(
             baseDirectory = baseDirectory,
             baseCacheDirectory = cacheDirectory,
             clock = clock,
+            deviceId = sessionData.deviceId,
```

Comment:
I would not add a constructor param here, but read the value from `client.deviceId()` to simplify even more (like it's done for [sessionId](https://github.com/element-hq/element-x-android/blob/main/libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClient.kt#L104)). Sorry if my first comment was not clear. Or there is a limitation that I don't see?

Reply:
Oh, I didn't realise `Client` from the Rust SDK had this value. Indeed, that's a lot easier.